actors/builtin/multisig/multisig_actor.go

```diff
@@ -114,25 +105,31 @@ type TxnIDParams struct {
 func (a *MultiSigActor) Approve(rt vmr.Runtime, params *TxnIDParams) *vmr.EmptyReturn {
 	rt.ValidateImmediateCallerType(builtin.CallerTypesSignable...)
 	callerAddr := rt.ImmediateCaller()
-	a._rtValidateSignerOrAbort(rt, callerAddr)
-	a._rtApproveTransactionOrAbort(rt, params.ID)
+	var st MultiSigActorState
+	rt.State().Transaction(&st, func() interface{} {
+		a.validateAuthorizedParty(rt, &st, callerAddr)
+		return nil
+	})
+	a.approveTransaction(rt, params.ID)
 	return &vmr.EmptyReturn{}
 }
 
 func (a *MultiSigActor) Cancel(rt vmr.Runtime, params *TxnIDParams) *vmr.EmptyReturn {
 	rt.ValidateImmediateCallerType(builtin.CallerTypesSignable...)
 	callerAddr := rt.ImmediateCaller()
-	a._rtValidateSignerOrAbort(rt, callerAddr)
 
-	h, st := a.State(rt)
-	txn := getPendingTxn(rt, st.PendingTxns, params.ID)
-	proposer := txn.Approved[0]
-	if proposer != callerAddr {
-		rt.AbortStateMsg("Cannot cancel another signers transaction")
-	}
+	var st MultiSigActorState
+	rt.State().Transaction(&st, func() interface{} {
+		a.validateAuthorizedParty(rt, &st, callerAddr)
+		txn := getPendingTxn(rt, st.PendingTxns, params.ID)
```

Comment:
Will conflict with https://github.com/filecoin-project/specs-actors/pull/39, I think this should go in first.

Reply:
thanks